Feature/byeongtaek

byeongtaek12 opened the pull request:
## 📌 관련 이슈
- Closes #

## ✨ 변경 사항
- Comment, Like, Follow 인증 로직 추가
- ShareGoal 기능 구현, 테스트 코드 작성


## 📷 Postman 
- 이미지 첨부
![캡처1](https://github.com/user-attachments/assets/221e3901-6a3d-4fbf-bd14-85216f5a74ff)
![캡처2](https://github.com/user-attachments/assets/e78b826d-763f-4e11-86a8-6234fc8772d3)
![캡처3](https://github.com/user-attachments/assets/2eb184ca-3736-4f3f-8556-eb4882eb8f46)
![캡처4](https://github.com/user-attachments/assets/ed5548a6-4aee-4f5d-8de8-5ebf6f89b9f1)
![캡처5](https://github.com/user-attachments/assets/6b0bf839-5f01-479c-9caa-87f47ff3dfdc)
![캡처6](https://github.com/user-attachments/assets/abc5de6b-62e9-4243-94b8-75d5f4bde02f)
![캡처7](https://github.com/user-attachments/assets/c2fd68f5-d1fd-4ffa-a678-fe82e403c8b6)
![캡처8](https://github.com/user-attachments/assets/48f803d4-97fb-4257-b52c-5b40a564587c)

## ✅ 체크리스트
- [ ] 관련 이슈에 연결됨
- [ ] 기능이 정상적으로 동작함
- [ ] 코드 리뷰를 받았음
- [ ] 불필요한 로그/코드 제거됨

## 💬 기타 참고 사항
- 테스트 방법, 추가 설명 등 필요한 경우 작성해 주세요.

byeongtaek12 committed:
[feat] : follow 구현, [test] : 테스트 코드 작성 (#6)
1. Follow 엔티티 클래스 구현
2. FollowController 클래스 구현
3. FollowService 인터페이스 구현
4. FollowServiceImpl 클래스 구현
5. Follow dto 클래스들 구현
6. FollowRepository 인터페이스 구현
7. FollowServiceImplTest 일단 단위테스트로 구현
기능 - (팔로우 요청, 대상의 팔로워 조회(page), 대상의 팔로잉 조회(page), 팔로우 단건 조회, 팔로우 취소)api 구현

byeongtaek12 committed:
[feat] : comment 구현, [test] : 테스트 코드 작성 (#7)
1. Comment 엔티티 클래스 구현
2. CommentController 클래스 구현
3. CommentService 인터페이스 구현
4. CommentServiceImpl 클래스 구현
5. Comment dto 클래스들 구현
6. CommentRepository 인터페이스 구현
7. CustomCommentRepository 인터페이스 구현
8. CustomCommentRepositoryImpl 클래스 구현
9. CommentServiceImplTest 일단 단위테스트로 구현
기능 - (댓글 생성, 댓글 목록 조회(cursor 기반 QueryDSL), 댓글 단건 조회, 댓글 수정, 댓글 삭제)api 구현

byeongtaek12 committed:
[feat] : like 구현, [test] : 테스트 코드 작성 (#8)
1. Like 엔티티 클래스 구현
2. LikeController 클래스 구현
3. LikeService 인터페이스 구현
4. LikeServiceImpl 클래스 구현
5. Like dto 클래스들 구현
6. LikeRepository 인터페이스 구현
7. LikeServiceImplTest 일단 단위테스트로 구현
기능 - (좋아요 생성, 좋아요 목록 조회(page), 좋아요 단건 조회, 좋아요 취소)api 구현

byeongtaek12 committed:
[feat] : ErrorCode 에러 추가
1. GOAL_NOT_FOUND 추가
2. FOLLOW_FORBIDDEN 추가

byeongtaek12 committed:
[style] 공백 추가

byeongtaek12 committed:
[feat] : GoalRepository 메서드 추가
1. 목표 아이디가 같고 상태가 활성된 목표가 필요해서 추가

byeongtaek12 committed:
[feat] : UserRepository 메서드 추가
1. 아이디가 같고 삭제되지 않은 유저가 필요해서 추가

byeongtaek12 committed:
Merge branch 'dev' into feature/byeongtaek

byeongtaek12 committed:
[refactor] : comment 리팩토링 (#7)
1. CommentPageResponseDto -> CommentCursorResponseDto로 클래스명변경
2. getComments 안에 nextCursor를 구하는 로직을 CommentCursorResponseDto의 정적 팩토리 메서드 안으로 넣음
3. CustomCommentRepostitoryImpl 안에 필요없는 로직 제거

byeongtaek12 committed:
[refactor] : GoalRepository 상속 추가
1. QueryDSL을 사용한 CustomShareGoalRepository 추가 상속

byeongtaek12 committed:
[Feat] : ShareGoal 관련 SuccessCode 추가

byeongtaek12 committed:
Merge branch 'dev' into feature/byeongtaek

# Conflicts:
#	src/main/java/org/example/lifechart/common/enums/ErrorCode.java
#	src/main/java/org/example/lifechart/domain/goal/repository/GoalRepository.java
#	src/main/java/org/example/lifechart/domain/user/repository/UserRepository.java

byeongtaek12 committed:
[feat] : aws 관련 값들 추가

byeongtaek12 committed:
[refactor] : CommentCursorResponseDto nextCursor 로직 수정 (#7)
1. 마지막 요소의 id -1 -> 마지막 요소의 id로 수정
2. 수정에 따른 테스트 코드도 수정
(마지막 요소의 id -1 인 부분이 빠지고 조회되는 오류가 있어서 마지막 요소의 id로 nextCursor값을 반환하게끔 수정하여 정상 작동 확인

byeongtaek12 committed:
[feat] : shareGoal 기능 구현 (#38)
1. ShareGoalController 클래스 구현
2. ShareGoalService 인터페이스 구현
3. ShareGoalServiceImpl 클래스 구현
4. ShareGoal dto 클래스들 구현
5. CustomShareGoalRepository 인터페이스 구현
6. CustomShareGoalRepositoryImpl 클래스 구현
7. ShareGoalServiceImplTest 일단 단위테스트로 구현
8. StringToCategoryConverter 클래스 구현
9. StringToShareConverter 클래스 구현
기능 - (공유 목표 조회, 특정 유저 공유 목표 조회)api 구현

ChangHoonS commented on src/main/resources/application.properties:
우선 여기서는 적당히 사용한거 같긴한데 properties에 들어있는 내용중 민감한 정보는 환경변수 처리해서
값을 공유해서 각자의 환경에 넣어주는게 맞는거 같다고 생각합니다.
또 제가 알기로 application secret properties가 있는걸로 알아서 차라리 그 파일에 적용시키는게 
github에 안올라가고 맞다고 생각하는 이 부분 우선 월요일날 한번 안그래도 여쭤보려고 합니다.

le951 replied:
방금 회의 하면서 이야기한 내용 정리합니다.


'jwt.secret.key' 부분은 확실히 놓칠 수 있겠네요. 
고민해보겠습니다.

일단 개발용 default는 놔두고,
EC2에서 실행 시에도 default 쓰려 하면 오류나는 방향으로 보겠습니다.

byeongtaek12 committed:
Merge branch 'dev' into feature/byeongtaek

byeongtaek12 committed:
[refactor] : ShareGoalCursorResponseDto 클래스 정적 메서드 로직 수정 (#38)
1. 컨텐츠가 없다면 nextCursor가 npe 발생해서 컨텐츠가 없다면 null 값을 넣어주는 로직 추가

rlqor9688 commented on src/main/java/org/example/lifechart/domain/comment/entity/Comment.java:
```diff
@@ -21,20 +25,18 @@ public class Comment extends BaseEntity {
+	@ManyToOne
+	@JoinColumn(nullable = false, name = "goal_id")
+	private Goal goal;
 	private String contents;
```
contents에
@column(nullable=false) 조건 필요할 것 같습니다!

byeongtaek12 replied:
반영하겠습니다

byeongtaek12 replied:
반영 완료했습니다

rlqor9688 commented on src/main/java/org/example/lifechart/domain/comment/service/CommentServiceImpl.java:
```diff
@@ -19,38 +24,41 @@
-	public CommentResponseDto createComment(Long goalId, CommentRequestDto commentRequestDto) {
+	public CommentResponseDto createComment(Long authId, Long goalId, CommentRequestDto commentRequestDto) {
 		// 로그인 유저 존재 여부 검증
+		User findedUser = validUser(authId);
 		// 목표 존재 여부 검증
+		Goal findedGoal = validGoal(goalId);
```
find의 과거형이 found네요!
→ findedUser, findedGoal를 foundUser / foundGoal 로 치환하면 의미가 더 명확해질 듯 합니다!
동작에 영향을 주는 요인은 아니어서 추후 업데이트를 고려해봐도 괜찮을 것 같아요!

byeongtaek12 replied:
아 그렇네요 이왕 고치는 거 고치겠습니다 👍

byeongtaek12 replied:
반영완료 했습니다

woneyee reviewed (approved):
공유범위와 카테고리 필터 조건에 따라 분기처리를 다 고민하셨네요 목표 쓴 사람이 내가 팔로우 한 사람인지 판단하기 위해서 따로 조인 사용하신 부분이 queryDSL로 처리하니 조건부 조인이 깔끔하게 잘 처리된 것 같습니다 코드리뷰하면서 배우고 가네요

byeongtaek12 committed:
[refactor] : 변수명 변경
1. finded -> found

byeongtaek12 committed:
[style] : 공백 추가

byeongtaek12 committed:
[refactor] : Comment 필드 수정
1. contents는 null이면 안 되기 때문에 nullable 추가

rlqor9688 commented on src/main/java/org/example/lifechart/domain/shareGoal/dto/response/ShareGoalResponseDto.java:
```diff
@@ -0,0 +1,41 @@
+			.nickname(goal.getUser().getNickname())
+			.title(goal.getTitle())
+			.category(goal.getCategory())
+			.targetAmount(goal.getTargetAmount())
+			.remainingPeriod(period)
+			.progressRate(goal.getProgressRate())
```
오전에 말씀 나눈 것처럼 progressRate 부분에서 Goal에서 갖고 오는 로직은 우선 삭제 부탁드려요!🙏

byeongtaek12 replied:
일단 goal에서 받지 않고 null을 넣어두도록 하겠습니다.

byeongtaek12 replied:
아 이 부분은 지워도 되겠네요 이해했습니다. 지우겠습니다.

byeongtaek12 replied:
반영 완료했습니다.

rlqor9688 commented on src/main/java/org/example/lifechart/domain/shareGoal/repository/CustomShareGoalRepositoryImpl.java:
```diff
@@ -0,0 +1,83 @@
+	}
+
+	@Override
+	public List<Goal> findByAuthIdAndUserId(Long authid, Long userId) {
```
authid가 다른 코드들에서는 authId로 들어간 것으로 보아 이 부분에서만 authid로 쓰신 것 같네요!
이것도 로직에 영향을 주지는 않아서, 참고만 해주시면 좋을 것 같습니다~ 수정한다면 아래 57번줄도 수정되어야 할 것 같아요!

byeongtaek12 replied:
오 그렇네요 수정하겠습니다

byeongtaek12 replied:
반영 완료했습니다

rlqor9688 commented on src/main/java/org/example/lifechart/domain/shareGoal/repository/CustomShareGoalRepositoryImpl.java:
```diff
@@ -0,0 +1,83 @@
+		QGoal goal = QGoal.goal;
+		QFollow follow = QFollow.follow;
+		return jpaQueryFactory
+			.selectFrom(goal)
+			.leftJoin(follow).on(
+				follow.receiver.id.eq(goal.user.id).and(follow.requester.id.eq(authid))
+			)
+			.where(
+				goal.user.id.eq(userId),
+				goal.share.eq(Share.ALL)
+					.or(follow.id.isNotNull().and(goal.share.eq(Share.FOLLOWER))),
+				goal.status.eq(Status.ACTIVE)
+			)
```
👍👍👍👍👍

byeongtaek12 committed:
[refactor] : ShareGoalResponseDto 정적 팩토리 메서드 로직 수정 (#38)
1. goal Entity 필드에서 progressRate를 삭제할 예정이라 일단 null로 처리
2. Period 타입을 그대로 반환하게 되면 사용자 입장에서 이게 얼마나 남았는 지 파악이 불가하여 직접 String format 처리하고 Period 타입에서 String 타입으로 변경

byeongtaek12 committed:
[refactor] : CustomShareGoalRepositoryImpl findByAuthIdAndUserId 메서드 매개변수명 수정 (#38)

byeongtaek12 committed:
[test] : CommentServiceImplTest테스트 코드 추가 (#7)
1. 빼먹은 게 있어서 로그인 유저 검증과 골 검증 추가

rlqor9688 reviewed (commented):
검토하고 피드백 남겼습니다! 고생하셨습니다!

rlqor9688 reviewed (approved):
피드백 확인해주신 내용 확인했습니다! 승인하겠습니다!
